components/activity/editor/d2l-activity-editor-main-collection.js

```diff
@@ -0,0 +1,73 @@
+// START custom component imports
+// END custom component imports
+import '@brightspace-ui/core/components/list/list.js';
+import '@brightspace-ui/core/components/list/list-item.js';
+import { css, LitElement } from 'lit-element/lit-element.js';
+import { customHypermediaElement, html } from '../../../framework/hypermedia-components.js';
+import { HypermediaLitMixin, observableTypes } from '../../../framework/hypermedia-lit-mixin.js';
+
+const rels = Object.freeze({
+	collection: 'https://activities.api.brightspace.com/rels/activity-collection',
+	item: 'item'
+});
+
+class ActivityEditorMainCollection extends HypermediaLitMixin(LitElement) {
+
+	static get properties() {
+		return {
+			items: { type: Array, observable: observableTypes.subEntities, rel: rels.item, route: [{observable: observableTypes.link, rel: rels.collection}] }
+		};
+	}
+
+	static get styles() {
+		return [ css`
+			.d2l-activity-collection-body {
+				margin: auto;
+				max-width: 1230px;
+				padding: 0 1.5rem;
+			}
+			.d2l-activity-collection-body-content {
+				max-width: 820px;
+				padding: 0 0.35rem;
+			}
+			.d2l-activity-collection-activities {
+				margin: 0 -1.5rem;
+				max-width: 881px;
+				padding: 0 0.05rem;
+			}
+			.d2l-activity-collection-list-actions {
+				align-items: baseline;
+				display: flex;
+				justify-content: space-between;
+				margin: 0.9rem 0;
+				max-width: 820px;
+				position: relative;
+			}
+		`];
+	}
+
+	constructor() {
+		super();
+		this.items = [];
+	}
+
+	render() {
+		return html`
+			<div class="d2l-activity-collection-body">
+				<div class="d2l-activity-collection-body-content">
+					<div class="d2l-activity-collection-list-actions">
+						<d2l-button primary>add Activity</d2l-button>
+						<div class="d2l-body-compact">Count: ${this.items.length}</div>
+					</div>
+				</div>
+				<div class="d2l-activity-collection-activities">
+					<d2l-list>
+						${this.items.map(href => html`<d2l-list-item>${href}</d2l-list-item>`)}
+					</d2l-list>
+				</div>
+			</div>
+		`;
+	}
+}
+
+customHypermediaElement('d2l-activity-editor-main-collection', ActivityEditorMainCollection, 'd2l-activity-editor-main-collection');

```

Comment:
Likewise with other comments, this should extend the tag `d2l-activity-editor-main` and resolve on learning-path or activity-collection
```suggestion
customHypermediaElement('d2l-activity-editor-main-collection', ActivityEditorMainCollection, 'd2l-activity-editor-main', [['activity-collection'], ['learning-path']]);

```

Reply:
Oh yeah forgot about that!